code/datums/datumvars.dm

```diff
@@ -1407,3 +1407,26 @@
 	if(href_list["listrefresh"])
 		debug_variables(locate(href_list["listrefresh"]))
 		return TRUE
+
+/client/proc/debug_global_variables(var_search as text)
+	set category = "Debug"
+	set name = "Debug Global Variables"
+
+	if(!check_rights(R_ADMIN|R_VIEWRUNTIMES))
+		to_chat(usr, "<span class='warning'>You need to be an administrator to access this.</span>")
+		return
+
+	var_search = trim(var_search)
+	if(!var_search)
+		return
+	if(!GLOB.can_vv_get(var_search))
+		return
+	switch(var_search)
+		if("vars")
+			return FALSE
+	log_and_message_admins("is debugging the Global Variables controller with the search term \"[var_search]\"")
```

Comment:
I would move this down to inside the check so it only fires when it actually succeeds.

Reply:
Everything below that is a "success", it just sometimes returns text/a number instead of opening a window, since you cant really view variables on those